File: Android/BarkCloud.Android/app/src/main/java/com/barkfluff/BarkCloud/files/ui/FormatUtils.kt
package com.barkfluff.BarkCloud.files.ui

import android.content.Context
import com.barkfluff.BarkCloud.R
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

fun formatSize(context: Context, bytes: Long): String = when {
    bytes < 1024 -> context.getString(R.string.files_size_b, bytes)
    bytes < 1024L * 1024 -> context.getString(R.string.files_size_kb, bytes / 1024.0)
    bytes < 1024L * 1024 * 1024 -> context.getString(R.string.files_size_mb, bytes / (1024.0 * 1024.0))
    else -> context.getString(R.string.files_size_gb, bytes / (1024.0 * 1024.0 * 1024.0))
}

fun formatChildCount(context: Context, count: Int): String = when {
    count == 0 -> context.getString(R.string.files_items_count_zero)
    count % 10 == 1 && count % 100 != 11 -> context.getString(R.string.files_items_count_one, count)
    count % 10 in 2..4 && count % 100 !in 12..14 -> context.getString(R.string.files_items_count_few, count)
    else -> context.getString(R.string.files_items_count_many, count)
}

private val dateFormat = SimpleDateFormat("dd.MM.yyyy", Locale.getDefault())

fun formatDate(epochMillis: Long): String = dateFormat.format(Date(epochMillis))

File: Ios/BarkCloud/BarkCloud/Features/Files/UI/FormatUtils.swift
import Foundation

enum FormatUtils {
    private static let dateFormatter: DateFormatter = {
        let f = DateFormatter()
        f.dateFormat = "dd.MM.yyyy"
        return f
    }()

    static func formatSize(_ bytes: Int64) -> String {
        if bytes < 1024 {
            return String(localized: "files_size_b").replacingOccurrences(of: "%lld", with: "\(bytes)")
        }
        let kb = Double(bytes) / 1024.0
        if kb < 1024 {
            return String(format: NSLocalizedString("files_size_kb", comment: ""), kb)
        }
        let mb = kb / 1024.0
        if mb < 1024 {
            return String(format: NSLocalizedString("files_size_mb", comment: ""), mb)
        }
        let gb = mb / 1024.0
        return String(format: NSLocalizedString("files_size_gb", comment: ""), gb)
    }

    static func formatChildCount(_ count: Int) -> String {
        if count == 0 { return String(localized: "files_items_count_zero") }
        let mod10 = count % 10
        let mod100 = count % 100
        let key: String
        if mod100 >= 11 && mod100 <= 14 {
            key = "files_items_count_many"
        } else if mod10 == 1 {
            key = "files_items_count_one"
        } else if mod10 >= 2 && mod10 <= 4 {
            key = "files_items_count_few"
        } else {
            key = "files_items_count_many"
        }
        return String(format: NSLocalizedString(key, comment: ""), count)
    }

    static func formatDate(_ date: Date) -> String {
        dateFormatter.string(from: date)
    }
}
